packages/component-library/src/components/plmg-slider/plmg-slider.tsx

```diff
@@ -0,0 +1,345 @@
+  }
+  /**
+   * Define a descriptive name for the slider
+   *
+   * Allowed values:
+   * - any string
+   *
+   * Required for accessibility and should be a unique and descriptive
+   *
+   */
+  @Prop() name: string;
+  @Watch('name')
+  validateName(newValue: string) {
+    if (newValue && typeof newValue !== 'string' && !this.name)
+      throw new Error('name must be a string');
+  }
+
+  /**
+   * Define a range of values
+   *
+   * Allowed values:
+   * - An array of with at least two items
+   *
+   * Must be a list of values with at least two items
+   * with the first and last items set min and max values
+   * additional values set additional marks and labels
+   * sort the string passed to component on the client
+   * component will not sort the array
+   */
+  @Prop() rangeValues: Array<number>;
+  Event: any;
+  @Watch('rangeValues')
+  validateRangeValues(newValue: Array<number>) {
+    if (!Array.isArray(newValue) || newValue.length < 2)
+      throw new Error('rangeValues must be an array with at least two items');
+  }
+
+  /**
+   * Define thumb label visibility
+   *
+   * Allowed values:
+   *  - true
+   *  - false
+   *
+   * Default: true
+   */
+  @Prop() thumbLabel: boolean = true;
+  @Watch('thumbLabel')
+  onThumbLabel(newValue: boolean) {
+    if (typeof newValue !== 'boolean')
+      throw new Error('thumb label must be boolean');
+  }
+
+  /**
+   * Define step
+   *
+   * Allowed values:
+   * - Any number
+   *
+   * Slider's value will increase or decrease by stepValue
+   *
+   * When step is not provided the stepValue is set to 1% of the range
+   */
+  @Prop() step: number;
+  @Watch('step')
+  validateStep(newValue: number) {
+    if (typeof newValue !== 'number' || newValue <= 0)
+      throw new Error('step must be a positive number');
+  }
+
+  /**
+   *
+   * Store value, min, max, trackwidth, inputfield and step states
+   *
+   * trackWidth stores the width of container after the component loads to calculate relative positions.
+   *
+   */
+
+  @State() min: number;
+  @State() max: number;
+  @State() trackWidth: number;
+  @State() stepValue: number;
+  @State() inputFieldValue: number;
+  @State() value: number;
+
+  private handleSliderChange(ev) {
+    this.updateValue(ev.target.value);
+  }
+
+  private updateValue(newValue) {
+    this.value = newValue;
+    if (this.inputFieldValue !== this.value) {
+      this.inputFieldValue = this.value;
+    }
+    this.valueUpdated.emit({ value: this.value });
+  }
+
+  @Event() valueUpdated: EventEmitter;
+
+  private getAllowedInputs() {
+    let inputs = [];
+    inputs.push(this.min);
+    while (inputs[inputs.length - 1] <= this.max) {
+      const PREV = inputs[inputs.length - 1];
+      // Correct for floating point errors by rounding when necessary
+      const NEXT =
+        Math.round((PREV + this.stepValue + Number.EPSILON) * 1000000) /
+        1000000;
+      // Escape valve in case of over rounding
+      if (!(NEXT > PREV)) return;
```

Comment:
This is a bit unreliable. For example, when entering value "1" into the text field it will remove focus and kick the user out of input.
Bug reported https://app.asana.com/0/1200489836971088/1202546970461155/f